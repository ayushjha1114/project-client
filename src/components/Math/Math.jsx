import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  first: PropTypes.number,
  second: PropTypes.number,
  operator: PropTypes.string,
  children: PropTypes.func,
};
const defaultProps = {
  first: 0,
  second: 0,
  operator: '',
  children: () => {},
};

class Math extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  calculate = (first, second, operator) => {
  /*     switch (operator) {
    case '+':
      return first + second;
    case '-':
      return first - second;
    case '*':
      return first * second;
    case '/':
      if (second !== 0) {
        return first / second;
      } else if (second === 0) {
        return 'infinity';
      }
    default:
      return 'Invalid Operation';
    } */
    if (operator === '+') {
      return first + second;
    }
    if (operator === '-') {
      return first - second;
    }
    if (operator === '*') {
      return first * second;
    }
    if (operator === '/' && second !== 0) {
      return first / second;
    }
    if (operator === '/' && second === 0) {
      return 'infinity';
    }
    return 'Invalid Operation';
  }

  render() {
    const {
      first,
      second,
      operator,
      children,
    } = this.props;
    const result = this.calculate(first, second, operator);
    return (
      <>
        { children(first, second, operator, result) }
      </>
    );
  }
}

Math.propTypes = propTypes;
Math.defaultProps = defaultProps;

export default Math;
