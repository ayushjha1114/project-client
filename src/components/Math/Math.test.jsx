import React from 'react';
import { shallow, mount } from 'enzyme';
import Math, { calculate } from './Math';

describe('Math', () => {
  test('should render correctly', () => {
    const component = shallow(<Math />);
    expect(component).toMatchSnapshot();
  });
  it('check the type of props', () => {
    const props = {
      first: 0,
      second: 0,
    };

    const component = mount(<Math {...props} />);
    expect(component.prop('first')).toEqual(0);
    expect(component.prop('second')).toEqual(0);
  });
  test('Adding 5+2 equal to 7 in calculate func', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(5, 2, '+')).toBe(7);
  });
  test('Subtacting 5-2 equal to 3 in calculate func', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(5, 2, '-')).toBe(3);
  });
  test('Multiplying 5*2 equal to 10 in calculate func', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(5, 2, '*')).toBe(10);
  });
  test('Dividing 5/2 equal to 10 in calculate func', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(5, 2, '/')).toBe(2.5);
  });
  test('Dividing 5/0 equal to infinity in calculate func', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(5, 0, '/')).toBe('infinity');
  });
  test('fake data 2?3 equal to Invalid Operation', () => {
    const component = shallow(<Math />);

    expect(component.instance().calculate(2, 3, '?')).toBe('Invalid Operation');
  });
});
