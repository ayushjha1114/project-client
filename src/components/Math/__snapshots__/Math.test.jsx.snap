// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Math
    first={0}
    operator=""
    second={0}
  >
    [Function]
  </Math>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
